### PR TITLE
Enable reading process environment variables in EventPipe

### PR DESCRIPTION
**Pull request by LakshanF**
Follows similar pattern as Mono's `EventPipe` implementation with support for Windows and Linux.

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp [hunk 1]
```diff
@@ -669,6 +669,46 @@ bool ep_rt_aot_spin_lock_release (ep_rt_spin_lock_handle_t *spin_lock)
     return false;
 }
 
+#ifndef HOST_WIN32
+#if defined(__APPLE__)
+#if defined (HOST_OSX)
+extern "C" {
+char ***_NSGetEnviron(void);
+}
```
Comment (elinor-fung) [inline src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp:677]: nit: I like the single line extern "C" function declaration, since there's only one

**Review by elinor-fung — APPROVED** [review]
> Follows similar pattern as Mono's EventPipe implementation with support for Windows and Linux.

Can you also call this out as another example in that issue you filed some time ago around sharing code in event pipe implementations?

**Comment by LakshanF** [comment]
> > Follows similar pattern as Mono's EventPipe implementation with support for Windows and Linux.
> 
> Can you also call this out as another example in that issue you filed some time ago around sharing code in event pipe implementations?

Updated #87069

File: src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp (hunk 1 above)
Comment (elinor-fung) [reply src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp]: Without the braces (but that is just me). Or indent if you prefer the multi-line with braces.

**Pull request merged**